### DISCUSSION TITLE
[TimeSeries Panel] duration based time series

### DISCUSSION THREAD

**Discussion by Willena** [grafana/grafana#57382] [Dashboards, Panels & Visualizations]
Would it be something acceptable to make the timeseries panel able to parse other type of timestamps (or other time related data) ?
I'm thinking of duration to display and compare similar events that occurred during a period of time.
Here is a visual to briefly show the idea. 
![image](https://user-images.githubusercontent.com/1786065/197065281-a69b1394-fa25-4f1b-a928-fa3593e09625.png)

This should not be too difficult to implement, as it would only be a different formatter (and maybe column type) for the X axis. 
Feedback from the Grafana time is very welcome on that question. If it has some interest, I'll try to make an implementation and a pull request.

**Comment by leeoniya** [comment]
we've been discussing use cases like this (time + offset) & time/time comparisons in the past couple weeks. something will happen around this use case in the not-too-distant future (probably post-v10), but it's not 100% clear yet what shape it will take (there are many details to iron out to make it powerful instead of a one-off, limited party trick).

**Comment by Willena** [comment]
Just to complete, in my use case, the data is already formatted as two columns (duration, value) in the database.
The only thing needed is the duration display format.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
